### PR TITLE
Add Custom JavaScript ruleset section

### PR DESCRIPTION
**Pull request by lbadenhop**
Thank you for your contribution to the Amplify-Central repo.

## Describe the changes

Enter a brief description of your changes to communicate to the reviewers what you changed and why.

## Deploy preview link

Please add the deploy preview link to the **specific page** that you've changed.

## Checklist for contributors

Before submitting this PR:

* Verify that all status checks have passed
* For more information on the Markdown linter rules, see [DavidAnson/markdownlint](https://github.com/DavidAnson/markdownlint)
* For first-time contributors, ensure that you have signed the [Axway CLA](https://cla.axway.com/)

### CODE REVIEW

**Comment by docsbuilder** [comment]
### <span aria-hidden="true">✅</span> Deploy preview is ready!

 |  Name | Link |
 |---------------------------------|------------------------|
 |<span aria-hidden="true">🔨</span> Latest commit | f3271df |
 |<span aria-hidden="true">😎</span> Deploy preview | http://PR-415.amplify-central.opendocs-builder.pcloud.axway.int |

File: content/en/docs/manage_compliance/compliance_axwaycli.md [hunk 1]
```diff
@@ -247,3 +247,63 @@ You can create the above `APISpecLintingRuleset` via the CLI as follows.
...
+
+You can write your own custom JavaScript functions if the core function is not enough for your custom ruleset.  
+
+To start the process of enabling custom Javascript function support, please submit a request on [support.axway.com](https://support.axway.com) with your Organization ID and list of email contacts who have permission to access the customer Javascript functions.
+
+### Amplify Limitations
+
+By default, Amplify’s “Compliance Profile” UI and CLI applies limitations on what kind of custom linting rulesets you can submit. These limitations are:
+* Cannot submit JavaScript rulesets or functions.
+– Rulesets are limited to JSON and YAML files.
```
Comment (jquick-axway) [inline content/en/docs/manage_compliance/compliance_axwaycli.md:261]: You're using a unicode '-' dash which won't turn into a bullet in HTML.
Copy this to VS Code and you'll what I'm talking about.

Replace with an ASCII dash and add 2 spaces to make it a sub-bullet. For example...
```markdown
* Cannot submit JavaScript rulesets or functions.
  - Rulesets are limited to JSON and YAML files.
```

File: content/en/docs/manage_compliance/compliance_axwaycli.md [hunk 2]
```diff
@@ -247,3 +247,63 @@ You can create the above `APISpecLintingRuleset` via the CLI as follows.
...
+* Amplify ruleset cannot “extend” another Amplify ruleset.
+* Amplify ruleset cannot reference another file via an “unapproved” URL.
+– For security reasons, Amplify blocks all URLs except for what’s on its “approved” URL list.
+– Raw file access to this “Amplify Extension” repository’s main branch is on the “approved” list.
```
Comment (jquick-axway) [inline content/en/docs/manage_compliance/compliance_axwaycli.md:265]: Same issue as above. Replace with ASCII dashes and add spaces to turn them into sub-bullets.

File: content/en/docs/manage_compliance/compliance_axwaycli.md [hunk 3]
```diff
@@ -247,3 +247,63 @@ You can create the above `APISpecLintingRuleset` via the CLI as follows.
...
+This “Amplify Extension” repository allows you to bypass most of these limitations.
+
+### Amplify Extension Restrictions
+
+Ruleset files are not allowed to reference other rulesets via “unapproved” URLs.
+– A URL to this repository’s main branch is on Amplify’s “approved” list.
```
Comment (jquick-axway) [inline content/en/docs/manage_compliance/compliance_axwaycli.md:271]: Another unicode dash that won't turn into a bullet. And first line is missing a bullet too. Should be...
```markdown
* Ruleset files are not allowed to reference other rulesets via “unapproved” URLs.
  - A URL to this repository’s main branch is on Amplify’s “approved” list.
* JavaScript files are not allowed to access the network or filesystem.
* JavaScript files cannot be obfuscated.
* JavaScript can only import/require-in modules that are available to the Spectral CLI.
```

File: content/en/docs/manage_compliance/compliance_axwaycli.md [hunk 4]
```diff
@@ -247,3 +247,63 @@ You can create the above `APISpecLintingRuleset` via the CLI as follows.
...
+* JavaScript files cannot be obfuscated.
+* JavaScript can only import/require-in modules that are available to the Spectral CLI.
+
+### Custom Rulesets and Functions
+
+Your custom Spectral ruleset files should be in this folder. Rulesets can be written in JSON, YAML, and JavaScript file formats. Your custom Spectral
+JavaScript linting functions should be added to the ./functions subdirectory. See the Spectral documentation on how to write “Custom Functions”.
+You can test your custom rulesets and files in this folder via the Spectral CLI.
+For example…
+spectral lint --ruleset example.yaml ./tests/openapi-v2.yaml
```
Comment (jquick-axway) [inline content/en/docs/manage_compliance/compliance_axwaycli.md:282]: Should add a code fence around the CLI command like this...
````markdown
```
spectral lint --ruleset example.yaml ./tests/openapi-v2.yaml
```
````

File: content/en/docs/manage_compliance/compliance_axwaycli.md [hunk 5]
```diff
@@ -247,3 +247,63 @@ You can create the above `APISpecLintingRuleset` via the CLI as follows.
...
+
+* JavaScript files are not allowed to bundle/rollup dependencies.
+
+Your file changes to this repository must be submitted via a GitLab merge request, which can only be merged by Axway after a review process.
+[- Axway reserves the right to not approve a merge request if it does not comply with the above “Amplify Extension Restrictions”. -]
```
Comment (jquick-axway) [inline content/en/docs/manage_compliance/compliance_axwaycli.md:287]: The "Axway reserves the right..." sentence is supposed to be an alert/warning box. It should probably be written in the markdown like the below with the `alert` tags...
```
{{< alert title="Note" color="primary" >}} Axway reserves the right to not approve a merge request if it does not comply with the above "Amplify Extension Restrictions".{{< /alert >}}
```

File: content/en/docs/manage_compliance/compliance_axwaycli.md [hunk 6]
```diff
@@ -247,3 +247,63 @@ You can create the above `APISpecLintingRuleset` via the CLI as follows.
...
+
+#### Access Custom Ruleset in Amplify
+
+To use a custom ruleset in this repository in Amplify, you’ll need to create a separate ruleset file which “links” to it as shown below. You’ll need to replace
+<your-repo-name> with the name of this repository and the example.yaml file name with the one you want to use in Amplify. (Tip: In GitLab, click on a file
```
Comment (jquick-axway) [inline content/en/docs/manage_compliance/compliance_axwaycli.md:292]: We should surround the `<your-repo-name>` with back ticks like the below or else it won't show in the docs due to the angle brackets.
```
`<your-repo-name>`
```

File: content/en/docs/manage_compliance/compliance_axwaycli.md [hunk 7]
```diff
@@ -247,3 +247,63 @@ You can create the above `APISpecLintingRuleset` via the CLI as follows.
...
+and then click the “Open Raw” button to get the raw file URL to use.)
+extends: ["https://git-ext.ecd.axway.com/amplify-extension/<your-repo-name>/-/raw/main/api-linting/example.yaml"]
```
Comment (jquick-axway) [inline content/en/docs/manage_compliance/compliance_axwaycli.md:294]: Put a code fence around this...
````
```yaml
extends: ["https://git-ext.ecd.axway.com/amplify-extension/<your-repo-name>/-/raw/main/api-linting/example.yaml"]
```
````

**Review by jquick-axway — APPROVED** [review]

**Comment by lbadenhop** [comment]
@robertengaxway, please review my edits by clicking on https://github.com/Axway/amplify-central/pull/415/commits/44f34da55977a78fe0d477172c9a4d5c4ffe083c. Also, see comments on lines #281 and 296.

File: content/en/docs/manage_compliance/compliance_axwaycli.md [hunk 8]
```diff
@@ -247,3 +247,70 @@ You can create the above `APISpecLintingRuleset` via the CLI as follows.
...
+    * A URL to this repository’s main branch is on Amplify’s approved list.
+* JavaScript files are not allowed to access the network or filesystem.
+* JavaScript files cannot be obfuscated.
+* JavaScript can only import/require-in modules that are available to the Spectral CLI.
+
+### Custom rulesets and functions
+
+Your custom Spectral ruleset files must be in this folder. Rulesets can be written in JSON, YAML, and JavaScript file formats. Your custom Spectral JavaScript linting functions must be added to the ./functions subdirectory. See the [Spectral documentation](https://docs.stoplight.io/docs/spectral/a781e290eb9f9-custom-functions) on how to write Custom Functions.
+
+You can test your custom rulesets and files in this folder via the Spectral CLI. For example:
```
Comment (lbadenhop) [inline content/en/docs/manage_compliance/compliance_axwaycli.md:282]: What folder?

File: content/en/docs/manage_compliance/compliance_axwaycli.md [hunk 9]
```diff
@@ -247,3 +247,70 @@ You can create the above `APISpecLintingRuleset` via the CLI as follows.
...
+
+* Ruleset files are not allowed to reference other rulesets via unapproved URLs.
+    * A URL to this repository’s main branch is on Amplify’s approved list.
+* JavaScript files are not allowed to access the network or filesystem.
+* JavaScript files cannot be obfuscated.
+* JavaScript can only import/require-in modules that are available to the Spectral CLI.
+
+### Custom rulesets and functions
+
+Your custom Spectral ruleset files must be in this folder. Rulesets can be written in JSON, YAML, and JavaScript file formats. Your custom Spectral JavaScript linting functions must be added to the ./functions subdirectory. See the [Spectral documentation](https://docs.stoplight.io/docs/spectral/a781e290eb9f9-custom-functions) on how to write Custom Functions.
```
Comment (lbadenhop) [inline content/en/docs/manage_compliance/compliance_axwaycli.md:280]: What folder?

File: content/en/docs/manage_compliance/compliance_axwaycli.md [hunk 10]
```diff
@@ -247,3 +247,70 @@ You can create the above `APISpecLintingRuleset` via the CLI as follows.
...
+
+{{< alert title="Note" color="primary" >}}JavaScript files are not allowed to bundle/rollup dependencies.{{< /alert >}}
+
+Your file changes to the Amplify Extension repository must be submitted via a GitLab merge request, which can only be merged by Axway after a review process.
+
+{{< alert title="Note" color="primary" >}}Axway reserves the right to not approve a merge request if it does not comply with the above [Amplify Extension restrictions](#amplify-extension-restrictions).{{< /alert >}}
+
+#### Access custom ruleset in Amplify
+
+To use a custom ruleset in the Amplify Extension repository, you must create a separate ruleset file that links to it, as shown below. Replace `<your-repo-name>` with the name of this repository and the `example.yaml` file name with the one you want to use in Amplify.
```
Comment (lbadenhop) [inline content/en/docs/manage_compliance/compliance_axwaycli.md:296]: What is "this repository"

**Comment by jquick-axway** [comment]
@lbadenhop, I moved this info out of the CLI doc and into its own doc.
I also made it clear this feature requires a GitLab repository and indicated what folders the custom files need to be added to.

FYI: The original markdown doc this info was copied from is a ReadMe file that will be in the customer's GitLab repository. So, the "this repository" and "this folder" made more sense in that context. But for this document, I tried to make it clearer which folder location in the repo files must be written to.

**Review by robertengaxway — COMMENTED** [review]
the changes look good

**Pull request merged**